okhttp/src/main/java/okhttp3/OkHttpClient.java

```diff
@@ -512,6 +522,21 @@ public Builder writeTimeout(long timeout, TimeUnit unit) {
     }
 
     /**
+     * Sets the default deadline timeout for new connections. A value of 0 means no timeout,
+     * otherwise values must be between 1 and {@link Integer#MAX_VALUE} when converted
+     * to milliseconds.
+     */
+    public Builder deadlineTimeout(long timeout, TimeUnit unit) {
```

Comment:
I don’t think this is the right name. In Okio we have specific words for deadlines and timeouts:
http://square.github.io/okio/1.x/okio/okio/Timeout.html

> Timeouts specify the maximum time to wait for a single operation to complete. Timeouts are typically used to detect problems like network partitions. For example, if a remote peer doesn't return any data for ten seconds, we may assume that the peer is unavailable.
> 
> Deadlines specify the maximum time to spend on a job, composed of one or more operations. Use deadlines to set an upper bound on the time invested on a job. For example, a battery-conscious app may limit how much time it spends pre-loading content.